src/wazuh_db/wdb_parser.c

```diff
@@ -5009,14 +5016,19 @@ int wdb_parse_global_update_connection_status(wdb_t * wdb, char * input, char *
         j_id = cJSON_GetObjectItem(agent_data, "id");
         j_connection_status = cJSON_GetObjectItem(agent_data, "connection_status");
         j_sync_status = cJSON_GetObjectItem(agent_data, "sync_status");
+        j_disconnected_time = cJSON_GetObjectItem(agent_data, "disconnected_time");
 
-        if (cJSON_IsNumber(j_id) && cJSON_IsString(j_connection_status) && cJSON_IsString(j_sync_status)) {
+        if (cJSON_IsNumber(j_id) &&
+            cJSON_IsString(j_connection_status) &&
+            cJSON_IsString(j_sync_status) &&
+            cJSON_IsNumber(j_disconnected_time)) {
             // Getting each field
             int id = j_id->valueint;
             char *connection_status = j_connection_status->valuestring;
             char *sync_status = j_sync_status->valuestring;
+            int disconnected_time = j_disconnected_time->valueint;
```

Comment:
Please, add a sanity check for this value. This should take a value different from zero only when the status to be set is `DISCONNECTED`.

Reply:
Changed according to what we discussed